internal/pkg/daemon/enricher/enricher.go

```diff
@@ -76,23 +79,37 @@ func (e *Enricher) Run() error {
 	e.logger.Info("Starting log-enricher on node: " + nodeName)
 
 	e.logger.Info("Connecting to local GRPC server")
-	conn, cancel, err := e.impl.Dial()
-	if err != nil {
-		return errors.Wrap(err, "connecting to local GRPC server")
-	}
-	defer cancel()
-	defer e.impl.Close(conn)
-	client := api.NewSecurityProfilesOperatorClient(conn)
 
-	metricsAuditIncClient, err := e.impl.MetricsAuditInc(client)
-	if err != nil {
-		return errors.Wrap(err, "create metrics audit client")
-	}
+	var (
+		conn                  *grpc.ClientConn
+		cancel                context.CancelFunc
+		metricsAuditIncClient api.SecurityProfilesOperator_MetricsAuditIncClient
+		recordSyscallClient   api.SecurityProfilesOperator_RecordSyscallClient
+	)
 
-	recordSyscallClient, err := e.impl.RecordSyscall(client)
-	if err != nil {
-		return errors.Wrap(err, "create syscall recording client")
+	if err := util.Retry(func() (err error) {
+		conn, cancel, err = e.impl.Dial()
+		if err != nil {
+			return errors.Wrap(err, "connecting to local GRPC server")
+		}
+		client := api.NewSecurityProfilesOperatorClient(conn)
+
+		metricsAuditIncClient, err = e.impl.MetricsAuditInc(client)
+		if err != nil {
+			return errors.Wrap(err, "create metrics audit client")
```

Comment:
Shouldn't we close conn and call cancel also here..

Reply:
Uh yeah, fixed as suggested :+1: